test/unittests/t_file_watcher_kqueue.cc

```diff
@@ -0,0 +1,69 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <gtest/gtest.h>
+
+#include <map>
+
+#include "file_watcher_kqueue.h"
+#include "logging.h"
+#include "util/pointer.h"
+#include "util/posix.h"
+
+typedef std::map<file_watcher::Event, int> Counters;
+
+class TestEventHandler : public file_watcher::EventHandler {
+ public:
+  explicit TestEventHandler(Counters* ctrs) : counters_(ctrs) {}
+  virtual ~TestEventHandler() {}
+
+  virtual bool Handle(const std::string& /*file_path*/,
+                      file_watcher::Event event,
+                      bool* clear_handler) {
+    (*counters_)[event]++;
+    *clear_handler = true;
+    return true;
+  }
+
+  Counters* counters_;
+};
+
+class T_FileWatcherKqueue : public ::testing::Test {
+ protected:
+  void SetUp() { counters_.clear(); }
+
+  Counters counters_;
+};
+
+TEST_F(T_FileWatcherKqueue, ModifyThenDelete) {
```

Comment:
It would be good if we can avoid the sleeping.  The first can perhaps be removed entirely?  The second and third one can become an endless loops on `counters.find()`?